knpu/episode3/Challenges/AddingAbstract/Coding.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Challenges\AddingAbstract;
+
+use KnpU\ActivityRunner\Activity\CodingChallenge\CodingContext;
+use KnpU\ActivityRunner\Activity\CodingChallenge\CorrectAnswer;
+use KnpU\ActivityRunner\Activity\CodingChallengeInterface;
+use KnpU\ActivityRunner\Activity\CodingChallenge\CodingExecutionResult;
+use KnpU\ActivityRunner\Activity\CodingChallenge\FileBuilder;
+use KnpU\ActivityRunner\Activity\Exception\GradingException;
+
+class Coding implements CodingChallengeInterface
+{
+    /**
+     * @return string
+     */
+    public function getQuestion()
+    {
+        return <<<EOF
+When Darth is browing all the different `DeathStar` models,
+we want to print out a little description that describes
+each one. Inside of that description, we want to include
+the "laser range", but it's different based on the model.
+Use an abstract method to return descriptions with the
```

Comment:
I want the user to create a `abstract protected function getLaserRange()` method in `AbstractDeathStar` then implement only that in `Deathstar` and `DeathstarII`. Then they can keep `getDescription` on AbstractDeathstar.

Maybe we should change the end of the text:

```md
Create an abstract method called `getLaserRange()` in `AbstractDeathstar` and then use
that in `getDescription()` to set the `$range` variable. Then, make sure each
Deathstar class has the proper range:

* `DeathStar` laser range = 500
* `DeathStarII` laser range = 900
```